src/protocolgame.cpp

```diff
@@ -440,9 +440,10 @@ void ProtocolGame::onRecvFirstMessage(NetworkMessage& msg)
 		return;
 	}
 
-	if (ip != boost::asio::ip::make_address(result->getString("session_ip"))) {
+	Connection::Address sessionIP = boost::asio::ip::make_address(result->getString("session_ip"));
+	Connection::Address localhost = boost::asio::ip::make_address("127.0.0.1");
+	if (sessionIP != localhost && ip != sessionIP) {
```

Comment:
```suggestion
	Connection::Address sessionIP = boost::asio::ip::make_address(result->getString("session_ip"));
	if (!sessionIP.is_loopback() && ip != sessionIP) {
```

Will work for both IPv4 and IPv6 localhost